Make a few adjustments

EyeCantCU opened the pull request:
The CPU input boost and simple thermal commits were picked from your kernel for the OnePlus 3 as they appear to work and to be applicable here. Since TSENS is not used by your thermal driver, there is no reason to call it anymore so I removed it's initialization. Furthermore, I disabled IO stats and add random as they are not really beneficial on MMC devices, and IO stats only serves the purpose of monitoring transfer speeds which is just using up resources that we do not need in a kernel intended for daily use. Disabling the CRC check also lead to higher speeds when testing with my modified version of your kernel as well. Finally, I added back Smartmax. It appeared as though you had tried to get it working right for Bacon at an earlier stage, but never could. DerRomtester seems to have successfully fixed it for use with your kernel and it has worked flawlessly for me over the months

Merry Christmas ;)

snitm committed:
block: disable entropy contributions for nonrot devices

Clear QUEUE_FLAG_ADD_RANDOM in all block drivers that set
QUEUE_FLAG_NONROT.

Historically, all block devices have automatically made entropy
contributions.  But as previously stated in commit e2e1a148 ("block: add
sysfs knob for turning off disk entropy contributions"):
    - On SSD disks, the completion times aren't as random as they
      are for rotational drives. So it's questionable whether they
      should contribute to the random pool in the first place.
    - Calling add_disk_randomness() has a lot of overhead.

There are more reliable sources for randomness than non-rotational block
devices.  From a security perspective it is better to err on the side of
caution than to allow entropy contributions from unreliable "random"
sources.

Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Jens Axboe <axboe@fb.com>

Conflicts:
	drivers/block/nbd.c
	drivers/block/null_blk.c
	drivers/block/nvme-core.c
	drivers/block/rsxx/dev.c
	drivers/block/skd_main.c
	drivers/block/zram/zram_drv.c
	drivers/md/bcache/super.c
	drivers/s390/block/scm_blk.c
	drivers/s390/block/xpram.c

Change-Id: I733f051d20a48f2afce68b7113fd9f1261be430b

Cl3Kener committed:
block: Disable IO Stats * I see no need to have this enabled. Will test to verify that it is not needed.

Signed-off-by: DerRomtester <dieb.stefan.96@gmail.com>

kerneltoast committed:
cpu_input_boost: Give fb_boost_nb the highest priority

CPU boosts should come before anything else in the framebuffer notifier
chain in order to speed up system resume.

Signed-off-by: Sultanxda <sultanxda@gmail.com>

GalaticStryder committed:
simple thermal: add support for polling TSENS sensor

VADC seems to very buggy on this device and causes a great mess with
thermal driver. Switch to TSENS for now and add a future switcher for
this.

Signed-off-by: Ícaro Hoff <icarohoff@gmail.com>

GalaticStryder committed:
simple thermal: add VADC/TSENS polling mode switcher

XO_THERM_PU2 is what thermal-engine wants for MSM8974 devices, although
TSENS is a great option, it's way too much erratic and forces the driver
to act a lot when it is not even needed.
Now, this simple (poor code) switcher can allow boths modes in the same
driver at user will to choose one over the other.

Signed-off-by: Ícaro Hoff <icarohoff@gmail.com>

kerneltoast commented on include/linux/blkdev.h:
```diff
@@ -432,10 +432,10 @@ struct request_queue {
 				 (1 << QUEUE_FLAG_STACKABLE)	|	\
 				 (1 << QUEUE_FLAG_SAME_COMP)	|	\
-				 (1 << QUEUE_FLAG_ADD_RANDOM))
+				 (0 << QUEUE_FLAG_ADD_RANDOM))
```
Let's go for the upstream solution: https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=b277da0a8a594308e17881f4926879bd5fca2a2d

kerneltoast commented on drivers/thermal/Kconfig:
```diff
@@ -120,13 +120,12 @@ config THERMAL_QPNP_ADC_TM
 	  absence of a system wide thermal monitoring entity. Monitors temperature
 	  via a specified ADC channel on the VADC device, and limits the max frequency
-	  of the CPU cores accordingly. The code is simple, allows for up to 8 throttling
+	  of the CPU cores accordingly. The code is simple, allows for multiple throttling
```
The extra zones aren't really useful on the OPX. My last throttle zone already forces the maxfreq down to 729 MHz, so I don't think we need more throttling steps after that.

kerneltoast commented on drivers/mmc/core/core.c:
```diff
@@ -73,7 +73,7 @@ static struct workqueue_struct *workqueue;
  * So we allow it it to be disabled.
  */
-bool use_spi_crc = 1;
+bool use_spi_crc = 0;
```
SPI mode isn't used for both our eMMC and SD card, so this is a no-op.

kerneltoast commented on drivers/cpufreq/cpu_input_boost.c:
```diff
@@ -241,6 +251,12 @@ static void cpu_ib_input_event(struct input_handle *handle, unsigned int type,
 	enum boost_status ib_status;
 	bool do_boost;
 
+	/* Anticipate device wake-up and boost early */
```
Boosting while the screen is off caused panics and cpufreq violations for me with MSM8974 kernels, probably because off-screen boosts can be active when the system suspends. This didn't crash my MSM8996 kernel, but it still resulted in cpufreq violations, which is why I've stopped using it entirely.

kerneltoast commented on drivers/cpufreq/Makefile:
```diff
@@ -10,6 +10,7 @@ obj-$(CONFIG_CPU_FREQ_GOV_USERSPACE)	+= cpufreq_userspace.o
 obj-$(CONFIG_CPU_FREQ_GOV_ONDEMAND)	+= cpufreq_ondemand.o
 obj-$(CONFIG_CPU_FREQ_GOV_CONSERVATIVE)	+= cpufreq_conservative.o
 obj-$(CONFIG_CPU_FREQ_GOV_INTERACTIVE)	+= cpufreq_interactive.o
+obj-$(CONFIG_CPU_FREQ_GOV_SMARTMAX)	+= cpufreq_smartmax.o
```
Performance and stability with Interactive are solid right now; I don't want to fix something that isn't broken. Tweaking the CPU governor is a nightmare, so I'd rather not touch this again :P

kerneltoast commented on arch/arm/mach-msm/oneplus/board-8974.c:
```diff
@@ -133,7 +133,6 @@ void __init msm8974_add_drivers(void)
 		msm_clock_init(&msm8974_rumi_clock_init_data);
 	else
 		msm_clock_init(&msm8974_clock_init_data);
-	tsens_tm_init_driver();
```
Although it's true that TSENS' readings are totally ignored by both userspace and the kernel with my setup, it's not doing anything bad. Who knows, some users might actually want to view TSENS readings for fun. I only disabled the TSENS driver in my MSM8996 kernel because GCC's stack protector detected stack corruption from inside one of its IRQ handlers.

kerneltoast commented on include/linux/blkdev.h:
```diff
@@ -432,10 +432,10 @@ struct request_queue {
 #define QUEUE_FLAG_SANITIZE    19	/* supports SANITIZE */
 
-#define QUEUE_FLAG_DEFAULT	((1 << QUEUE_FLAG_IO_STAT) |		\
+#define QUEUE_FLAG_DEFAULT	((0 << QUEUE_FLAG_IO_STAT) |		\
```
I need to investigate this a bit more to find out exactly what it does in order to feel comfortable disabling it.

EyeCantCU commented:
Okay, so I removed changes that were questionable. I do believe though that CPU input boost should have boost's with the maximum priority so as to avoid conflict, and while we can leave TSENS enabled, it does not need to be a dependency of your simple thermal driver. I'm keeping the IO stats change as we really only need this for storage benchmarks, otherwise it is incredibly pointless. I replaced the commit where add random was disabled with the upstream implementation as well

EyeCantCU commented:
Another thing to consider... I have CFQ working perfectly on my implementation of your kernel without the random reboots or any of that, and it has been doing well for the past day or so. The only issue is that my MMC implementation requires backporting changes from 3.10 which I believe is something that you wouldn't want to do on your part. It works fine, but I just wanted to let you know that it is still a possibility if you don't mind the potential offsets ensued by using the MMC implementation from 3.10

EyeCantCU commented:
Made a few more changes... Let's keep the TSENS dependency and give the user the ability to decide between VADC or TSENS for the thermal driver :)

Changes to IO stats and entropy contributions still persist

EyeCantCU commented:
Forget what I said about CFQ... I now see what you mean. It took a lot of time, but I see what you mean :). After taking a log, it appears to still be correlated with our MMC device... how unfortunate. Back to deadline or zen it is...

EyeCantCU commented:
Meh. Two different mindsets with two different visions. Thank you for your time Sultan. I appreciate it
